pkg/predict/predictor.go

```diff
@@ -131,18 +132,45 @@ func (p *Predictor) Predict(inputs Inputs) (*Response, error) {
 	}
 	defer resp.Body.Close()
 
-	// TODO
-	if resp.StatusCode == http.StatusBadRequest {
+	if resp.StatusCode == http.StatusUnprocessableEntity {
 		body := struct {
-			Message string `json:"message"`
+			Detail []struct {
+				Location []string `json:"loc"`
+				Message  string   `json:"msg"`
+				Type     string   `json:"type"`
+			} `json:"detail"`
```

Comment:
This struct could be defined at the top level so it's reuseable, like `Request` and `Response`. This isn't important nor essential, but Go code tends to do that except for trivial types, because as soon as you want to define a function that needs to take/return this type, then it needs to be defined somewhere. It also will then show up in compile errors and your editor as `ErrorResponse` or whatever, rather than some cryptic unnamed type.

Also, another consideration -- I expect we'll use reuse struct for other error responses. It's a generic structure for FastAPI errors, I think.

Anyway -- small, non-important thing, but reviewing this intentionally verbosely for information transfer. 😄